### PR TITLE
Update to Go v1.18

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -136,3 +135,4 @@ linters:
 #    - wastedassign # wastedassign finds wasted assignment statements. [fast: false, auto-fix: false]
 #    - wrapcheck # Checks that errors returned from external packages are wrapped [fast: false, auto-fix: false]
 #    - wsl # Whitespace Linter - Forces you to use empty lines! [fast: true, auto-fix: false]
+#    - ifshort # Checks that your code uses short syntax for if-statements whenever possible [fast: true, auto-fix: false]
```
Comment (andreaangiolillo) [inline .golangci.yml:138]: this lint has been deprecated

File: mongodbatlas/doc.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ pass cancellation signals and deadlines to various services of the client for
 handling a request. In case there is no context available, then context.Background()
 can be used as a starting point.
 
-Authentication
+Authentication.
```
Comment (blva) [inline mongodbatlas/doc.go:36]: [q] is this a linter requirement?

**Review by blva — DISMISSED** [review]

File: mongodbatlas/doc.go (hunk 2 above)
Comment (andreaangiolillo) [reply mongodbatlas/doc.go:36]: yes, the lint test was failing because of this

Comment (blva) [reply mongodbatlas/doc.go:36]: I think this is a bit odd. For example, we don't need the dot in line 18. Could you share the rule that requires this? Wondering if we can remove it/search if it's a bug/try to tweak the text to not require this...

Comment (andreaangiolillo) [reply mongodbatlas/doc.go:36]: you are right. I will try to remove it

Comment (andreaangiolillo) [reply mongodbatlas/doc.go]: Screenshot 2022-08-16 at 10 10 57

**Review by blva — APPROVED** [review]

**Pull request merged**